Configuration/src/main/java/io/deephaven/configuration/ConfigDir.java

```diff
@@ -66,6 +64,14 @@ private static Optional<String> configDirectoryFileIfExists() {
                 .map(Path::toString);
     }
 
+    private static Optional<String> getConfigDirOptional() {
```

Comment:
It bothers me that you called this `getOptional` in `CacheDir` and `DataDir` and a more verbose name here. Pick one style for consistency.